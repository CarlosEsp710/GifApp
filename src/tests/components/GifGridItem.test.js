import React from "react";
import { shallow } from "enzyme";

import { GifGridItem } from "../../components/GifGridItem";

describe("GifGridItem", () => {
  const title = "Title";
  const url = "url";
  let wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should return component <GifGridItem />"', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have a <p></p> with the title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("should have a <img/> with the src and alt from the props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("should have className animate__fadeIn", () => {
    const div = wrapper.find("div");
    expect(div.prop("className").includes("animate__fadeIn")).toBe(true);
  });
});
