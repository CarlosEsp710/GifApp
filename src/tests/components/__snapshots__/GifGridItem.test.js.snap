// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem should return component <GifGridItem />" 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-1s"
>
  <img
    alt="Title"
    src="url"
  />
  <p>
    Title
  </p>
</div>
`;
